src/extractor/guidance/turn_handler.cpp

```diff
@@ -459,14 +522,62 @@ Intersection TurnHandler::assignRightTurns(const EdgeID via_edge,
     return intersection;
 }
 
-std::pair<std::size_t, std::size_t> TurnHandler::findFork(const EdgeID via_edge,
-                                                          const Intersection &intersection) const
+// given two adjacent roads and `road1` being a candidate for a fork,
+// return false, if next road `road2` is also a fork candidate or
+// return true, if `road2` is not a suitable fork candidate and thus, `road1` the outermost fork
+bool isOutermostForkCandidate(const ConnectedRoad &road1, const ConnectedRoad &road2)
 {
+    const auto angle_between_next_road_and_straight = angularDeviation(road2.angle, STRAIGHT_ANGLE);
+    const auto angle_between_prev_road_and_next = angularDeviation(road1.angle, road2.angle);
+    const auto angle_between_prev_road_and_straight = angularDeviation(road1.angle, STRAIGHT_ANGLE);
+
+    // a road is a fork candidate if it is close to straight or
+    // close to a street that goes close to straight
+    // (reverse to find fork non-candidate)
+    if (angle_between_next_road_and_straight > NARROW_TURN_ANGLE)
+    {
+        if (angle_between_prev_road_and_next > NARROW_TURN_ANGLE ||
+            angle_between_prev_road_and_straight > GROUP_ANGLE)
+        {
+            return true;
+        }
+    }
+    return false;
+}
+
+template <typename Iterator>
+std::size_t findOutermostForkCandidate(Iterator base, Iterator start, Iterator end)
+{
+    const auto outermost_fork_candidate = std::adjacent_find(start, end, isOutermostForkCandidate);
+    auto outermost_fork_candidate_index = std::distance(base, outermost_fork_candidate);
+    // if Iterator is a reverse iterator, the index is different
+    if (outermost_fork_candidate_index < 0)
+    {
+        outermost_fork_candidate_index = std::abs(outermost_fork_candidate_index) - 1;
+    }
+
+    if (outermost_fork_candidate != end)
+    {
+        return outermost_fork_candidate_index;
+    }
+    // if all roads are part of a fork, set `candidate` to the last road
+    else
+    {
+        return outermost_fork_candidate_index - 1;
+    }
+}
 
+// @CODE_REVIEW_QUESTION: tuple or struct? I cannot decide
+// returns a tuple {i, angle_dev} of the road at the intersection that is closest to going straight:
+//     `intersection[i]` is the actual road
+//     `angle_dev` is the angle between going straight and taking the road `intersection[i]`
+std::tuple<std::size_t, double>
+TurnHandler::findClosestToStraight(const Intersection &intersection) const
```

Comment:
An Intersection always has the u-turn. Finding the closest to straight in such an intersection would be the u-turn so I think the return type should be ok and not optional.

Probably it would be good to have some assertions that Intersections always have the u-turn road but shouldn't be checked in this function in my opinion?